File: src/chipschallenge/blockreactions/BrownButtonFrom.java
package chipschallenge.blockreactions;

import chipschallenge.Block;
import chipschallenge.Buttons;

/**
 *
 * @author patrik
 */
public class BrownButtonFrom extends BlockReaction {

    private BrownButtonFrom() {}
    private static BrownButtonFrom mInstance = null;
    public static synchronized BrownButtonFrom getInstance() {
        if(mInstance == null)
            mInstance = new BrownButtonFrom();
        return mInstance;
    }

    public void react(Block moving, Block standing) {
        Buttons.brownButtonUp(standing);
    }

    public boolean canMove(Block moving, Block standing) {
        return true;
    }

}

File: ChipsChallenge.Shared/Blockreactions/BrownButtonFrom.cs
﻿namespace ChipsChallenge.Shared.Blockreactions
{
    public class BrownButtonFrom : NoSlipReaction
    {
        private static BrownButtonFrom instance;

        private BrownButtonFrom()
        {
        }

        public static BrownButtonFrom Instance
        {
            get
            {
                lock (typeof(BrownButtonFrom))
                {
                    return instance ?? (instance = new BrownButtonFrom());
                }
            }
        }

        public override void React(Block moving, Block standing)
        {
            Buttons.BrownButtonUp(standing);
        }

        public override bool canMove(Block moving, Block standing)
        {
            return true;
        }
    }
}